src/app/campaign/store/selectors/campaign-flight.selectors.ts

```diff
@@ -69,3 +76,85 @@ export const selectRoutedCampaignFlightDocs = createSelector(
     return { campaignDoc: campaign.doc, flightDoc: flight.doc };
   }
 );
+
+export const selectCampaignFlightInventoryReportData = createSelector(
+  selectCampaign,
+  selectAllInventory,
+  selectAllAdvertisers,
+  selectRemoteFlightsOrderByContractStartAt,
+  selectFlightDaysEntities,
+  (campaign: CampaignState, shows: Inventory[], advertisers: Advertiser[], flights: Flight[], inventory: Dictionary<FlightDays>) => {
+    // Flight Names
+    const reportData: any[][] = [['Flight Name'].concat(flights.map(flight => flight.name))];
+    // Show
+    reportData.push(
+      ['Show'].concat(
+        flights.map(flight => {
+          const show = shows.find(i => i.self_uri === flight.set_inventory_uri);
+          return show ? show.podcastTitle : '';
+        })
+      )
+    );
+    // Advertiser/Sponsor
+    reportData.push(
+      ['Sponsor'].concat(
+        flights.map(_ => {
+          const advertiser = advertisers.find(a => a.set_advertiser_uri === campaign.remoteCampaign.set_advertiser_uri);
+          return advertiser ? advertiser.name : '';
+        })
+      )
+    );
+    // Zones
+    reportData.push(['Zones'].concat(flights.map(flight => flight.zones.map(zone => zone.label).join('|'))));
+    // Contract Start and End Dates
+    reportData.push(
+      ['Start Date'].concat(flights.map(flight => (flight.contractStartAt ? flight.contractStartAt.format('MM-DD-YYYY') : '')))
```

Comment:
Yeah, and same here - I think these should fallback to the non-contract start/end dates.

Reply:
Same question here about falling back to both actual dates if the contract start or end date is missing, or fallback independently.

Reply:
I was thinking fallback independently.

Per the last ad-ops meeting ... I do think we'll start defaulting `f.contractStartAt = f.startAt` and `f.contractEndAt = f.endAt` on the frontend at some point.  Which sort of makes them non-optional fields.  But doesn't hurt to belt-and-suspenders to do the fallback here.